xrviz/projection.py

```diff
@@ -0,0 +1,113 @@
+import inspect
+import panel as pn
+import xarray as xr
+import geoviews as gv
+from geoviews import tile_sources as gvts
+from cartopy import crs as ccrs
+from .sigslot import SigSlot
+
+projections = [p for p in dir(ccrs) if inspect.isclass(getattr(ccrs, p)) and issubclass(getattr(ccrs, p), ccrs.Projection) and not p.startswith('_')]
+not_to_include = ['Projection', 'UTM']
+projections_list = sorted([p for p in projections if  p not in not_to_include ])
+
+
+class Projection(SigSlot):
+
+    def __init__(self):
+        super().__init__()
+        self.is_geo = pn.widgets.Checkbox(name='is_geo', value=False,
+                                          disabled=True)
+        self.alpha = pn.widgets.FloatSlider(name='alpha', start=0, end=1,
+                                            step=0.01, value=0.7, width=180)
+
+        self.show_map = pn.widgets.Checkbox(name='show_map', value=False,
+                                            width=100)
+        self.basemap = pn.widgets.Select(name='basemap',
+                                         options=gvts.tile_sources,
+                                         value=gvts.OSM,
+                                         disabled=True)
+        self.projection = pn.widgets.Select(name='projection',
+                                            options=projections_list,
+                                            value='PlateCarree')
+        self.crs = pn.widgets.Select(name='crs',
+                                     options=[None] + sorted(projections_list),
+                                     value=None)
+        self.rasterize = pn.widgets.Checkbox(name='rasterize', value=True)
+        self.project = pn.widgets.Checkbox(name='project', value=True)
+        self.global_extent = pn.widgets.Checkbox(name='global_extent',
+                                                 value=False)
+        self.proj_params = pn.Row()
+
+        feature_ops = ['None', 'borders', 'coastline', 'grid', 'land', 'lakes',
+                       'ocean', 'rivers']
+        self.features = pn.widgets.MultiSelect(name='features',
+                                               options=feature_ops,
+                                               value=feature_ops[1:])
+
+        self._register(self.is_geo, 'is_geo_value')
```

Comment:
Prefer the names to be verbs, since they represent actions, so "geo_changed" in this case, "geo_disabled" for the next.